components/script/dom/pannernode.rs

```diff
@@ -54,16 +54,29 @@ impl PannerNode {
     ) -> Fallible<PannerNode> {
         let count = options.parent.channelCount.unwrap_or(2);
         let mode = options.parent.channelCountMode.unwrap_or(ChannelCountMode::Clamped_max);
+        let interpretation = options.parent.channelInterpretation.unwrap_or(ChannelInterpretation::Speakers);
         if mode == ChannelCountMode::Max {
             return Err(Error::NotSupported)
         }
-        if count > 2 {
+        if count > 2 || count == 0 {
             return Err(Error::NotSupported)
         }
+        if *options.maxDistance <= 0. {
+            return Err(Error::Range("maxDistance should be positive".into()))
+        }
+        if *options.refDistance < 0. {
```

Comment:
Perhaps `<= 0.`. The spec says to throw when it is set to a non-positive value. Same as with `maxDistance`